refactor: Remove redundant conditional

Bushstar opened the pull request:
Conditional check against fMaster is now redundant as it is already checked as true. This originally made sense as the outer conditional was:

https://github.com/bitcoin/bitcoin/blob/f9cae832e6f56c6abe89b3bf05d1f176c2a7c913/src/checkqueue.h#L86

Removal of fQuit happened in the commit below.

https://github.com/bitcoin/bitcoin/commit/30ded3e3d888f75b6fa8b2e55a3883f799e00775#diff-88316c9aa9514c038c9304297e672da5

practicalswift commented:
ACK 647c63c4f3ffe9fb3b5ff091f9e85dc4667e54ee -- redundant code is confusing and obscures the intent of the code

Thanks for linking to the removal to make the context clear.

Bushstar commented:
@fanquake the build failed to what seems like an issue unrelated to this commit, same reason the following builds failed on recent commits.

https://ci.appveyor.com/project/DrahtBot/bitcoin/builds/30242431
https://ci.appveyor.com/project/DrahtBot/bitcoin/builds/30242115

hebasto commented:
Is there a unit test for that?

kristapsk reviewed (approved):
ACK 647c63c4f3ffe9fb3b5ff091f9e85dc4667e54ee

emilengler commented:
ACK 647c63c

Maybe it's better to change the title of the PR and the commit mesage to:
```
refactor: Remove redundant conditional
```

Bushstar committed:
refactor: Remove redundant conditional

Bushstar commented:
> ACK [647c63c](https://github.com/bitcoin/bitcoin/commit/647c63c4f3ffe9fb3b5ff091f9e85dc4667e54ee)
> 
> Maybe it's better to change the title of the PR and the commit mesage to:
> 
> ```
> refactor: Remove redundant conditional
> ```

Sure, maybe it'll pass CI this time, something seems amiss with appveyor.

theStack reviewed (approved):
ACK 647c63c

promag reviewed (commented):
ACK e80317be5fc6f6a04ea3b35bfe9991b3a5d29f7a.

Removed condition already checked at line 89.

Restarted appveyor.

emilengler commented:
re-ACK e80317be5fc6f6a04ea3b35bfe9991b3a5d29f7a

practicalswift commented:
ACK e80317be5fc6f6a04ea3b35bfe9991b3a5d29f7a

hebasto commented:
ACK e80317be5fc6f6a04ea3b35bfe9991b3a5d29f7a, I have reviewed the code, and it looks OK, I agree it can be merged.

Bushstar commented:
Trying again now that the appveyor issue appears to be fixed.

Bushstar commented:
@fanquake Trying again now that the appveyor issue appears to be fixed.

Note that commit is still https://github.com/bitcoin/bitcoin/commit/e80317be5fc6f6a04ea3b35bfe9991b3a5d29f7a, just changed and changed back to rerun CI.

theStack commented:
ACK https://github.com/bitcoin/bitcoin/commit/e80317be5fc6f6a04ea3b35bfe9991b3a5d29f7a

Empact commented:
ACK https://github.com/bitcoin/bitcoin/pull/17971/commits/e80317be5fc6f6a04ea3b35bfe9991b3a5d29f7a
